capsules/src/net/ipv6/ip_utils.rs

```diff
@@ -0,0 +1,213 @@
+use net::icmpv6::icmpv6::{ICMP6Header, ICMP6HeaderOptions};
+use net::ipv6::ipv6::IP6Header;
+use net::udp::udp::UDPHeader;
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum MacAddr {
+    ShortAddr(u16),
+    LongAddr([u8; 8]),
+}
+
+pub mod ip6_nh {
+    pub const HOP_OPTS: u8 = 0;
+    pub const TCP: u8 = 6;
+    pub const UDP: u8 = 17;
+    pub const IP6: u8 = 41;
+    pub const ROUTING: u8 = 43;
+    pub const FRAGMENT: u8 = 44;
+    pub const ICMP: u8 = 58;
+    pub const NO_NEXT: u8 = 59;
+    pub const DST_OPTS: u8 = 60;
+    pub const MOBILITY: u8 = 135;
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct IPAddr(pub [u8; 16]);
+
+impl IPAddr {
+    pub fn new() -> IPAddr {
```

Comment:
OK to have this, but it probably shouldn't actually be used anywhere... Just require the user to pass in an `IPAddr` _always_ instead of defaulting to a meaningless address (they'll have to specify it at some point anyway).